### ISSUE TITLE
2.1 catalog filter displaying below grid

### ISSUE THREAD

**Issue by maderlock** [magento/magento2#5408]
## Steps to reproduce
1. 2.0.7 upgraded to 2.1.0 running on PHP 7.0.7
2. Plenty of products added
3. Visit admin product catalog page
## Expected result
1. In 2.0.7 the filters were above the grid
## Actual result
1. They are now displaying below the grid. Screenshot: 
   ![screenshot 2016-07-01 16 43 11](https://cloud.githubusercontent.com/assets/630320/16527048/624ad8b0-3fac-11e6-83f6-99ca7af06136.png)

If there is a full page of products in the grid then they cannot be seen at all. The fixed tool-bar section still is shown when scrolling, but clicking of the filter icon does not make it appear.

I note that the knockoutjs template has the filter after the grid, so something should be rearranging it. There are no javascript errors, so I do not know what might be causing this.

I don't know if the above instructions will be sufficient to reproduce, but I don't know what else is relevant.

**Comment by maderlock** [comment]
Having added a column to the order grid as well I'm getting the same issue, so this seems to be a function of custom columns. Is there any known cure?

**Comment by reijervano** [comment]
Does anyone knows a workaround?

**Comment by kanduvisla** [comment]
I'm running in this problem too. I try to add a custom column to the order grid with my custom `view/adminhtml/ui_component/sales_order_grid.xml`:

```
<?xml version="1.0" encoding="UTF-8"?>
<listing xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Ui:etc/ui_configuration.xsd">
    <columns name="sales_order_columns">
        <column name="order_reference">
            <argument name="data" xsi:type="array">
                <item name="config" xsi:type="array">
                    <item name="dataType" xsi:type="string">text</item>
                    <item name="visible" xsi:type="boolean">true</item>
                    <item name="label" xsi:type="string" translate="true">Order Reference</item>
                </item>
            </argument>
        </column>
    </columns>
</listing>
```

Doing this moves the filters to the bottom. Any thoughts on this?

**Comment by kanduvisla** [comment]
Small update: it probably has something to with the order how UI components are rendered. If you add an empty `<listingToolbar name="listing_top"/>` above your `<columns>`-tag, it works. My main guess is because this way the element will already be created in the javascript-array that renders the HTML.

It might also have something to do with a ticket I filed a while back which shows an issue that happens with how various XML files get merged into one: https://github.com/magento/magento2/issues/8647

**Comment by magento-engcom-team** [comment]
@maderlock, thank you for your report.
The issue is already fixed in develop branch, 2.2.0, 2.1.9

**Comment by meezin** [comment]
@magento-engcom-team Hi there! It doesn't fix in 2.1.9

**Comment by fauzie** [comment]
@magento-engcom-team It doesn't fix also in 2.2.0
